### PR TITLE
[Code Together] Add a deprecation warning for `qml.Interferometer` before renaming

### PR DESCRIPTION
**Pull request by AkashNarayanan**
**Context:**
 Add a deprecation warning for `qml.Interferometer` and a test case to check the warning.

**Description of the Change:**
- Added a `UserWarning` with a deprecation message to the `ops.cv.Interferometer` class explaining that it will be renamed to `InterferometerUnitary`.
- Added a test case to check if a warning is raised when creating the operation.

**Benefits:**
Informs users about the deprecation of `qml.Interferometer`.

**Related GitHub Issues:**
#1531

### CODE REVIEW

**Comment by thisac** [comment]
Thanks @AkashNarayanan for working on this! We'll have a look at the PR and review it soon.

**Review by thisac — CHANGES_REQUESTED** [review]
Thanks @AkashNarayanan! Just a small comment on the formatting. Looks great otherwise!

File: pennylane/ops/cv.py [hunk 1]
```diff
@@ -38,6 +38,7 @@
 import math
 import numpy as np
 from scipy.linalg import block_diag
+import warnings
```
[the remark this answers is not in the document]
Comment (AkashNarayanan) [reply pennylane/ops/cv.py]: Thank you for the feedback @thisac ! I have already changed the import order and have pushed it to the branch (39e5247ec7708433af29548f4f51b5709402963f). I will make the remaining suggested changes and push them soon.

**Comment by AkashNarayanan** [comment]
Hi @thisac ! I have made the suggested changes.

**Comment by thisac** [comment]
Hi @AkashNarayanan! Thanks for making the changes. Looks good, so I'll approve and merge. 💯

**Review by thisac — APPROVED** [review]

**Pull request merged**